tests/test_generators/test_graphql.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+from linkml.generators.graphqlgen import GraphqlGenerator
+
+PERSON = """
+type Person implements HasAliases
+  {
+    id: String!
+    name: String
+    hasEmploymentHistory: [EmploymentEvent]
+    hasFamilialRelationships: [FamilialRelationship]
+    hasMedicalHistory: [MedicalEvent]
+    ageInYears: Integer
+    addresses: [Address]
+    hasBirthEvent: BirthEvent
+    speciesName: String
+    stomachCount: Integer
+    isLiving: LifeStatusEnum
+    aliases: [String]
+  }
+"""
+
+MEDICALEVENT = """
+type MedicalEvent
+  {
+    startedAtTime: Date
+    endedAtTime: Date
+    isCurrent: Boolean
+    metadata: AnyObject
+    inLocation: Place
+    diagnosis: DiagnosisConcept
+    procedure: ProcedureConcept
+  }
+"""
+
+FAMILIALRELATIONSHIP = """
+type FamilialRelationship
+  {
+    startedAtTime: Date
+    endedAtTime: Date
+    cordialness: String
+    type: FamilialRelationshipType!
+    relatedTo: Person!
+  }
+"""
+
+DATASET = """
+type Dataset
+  {
+    metadata: AnyObject
+    persons: [Person]
+    companies: [Company]
+    activities: [Activity]
+    codeSystems: [CodeSystem]
+  }
+"""
+
+FAMILIALRELATIONSHIPTYPE = """
+enum FamilialRelationshipType
+  {
+    SIBLING_OF
+    PARENT_OF
+    CHILD_OF
+  }
+"""
+
+
+@pytest.mark.parametrize(
+    "input_class,expected",
+    [
+        # check that expected GraphQL schema blocks are present
+        ("Person", PERSON),
+        ("Dataset", DATASET),
+        ("MedicalEvent", MEDICALEVENT),
+        ("FamilialRelationship", FAMILIALRELATIONSHIP),
+        ("FamilialRelationshipType", FAMILIALRELATIONSHIPTYPE),
+    ],
+)
+def test_serialize_selected(input_class, expected, kitchen_sink_path):
+    """Test serialization of select GraphQL schema from schema."""
+    generator = GraphqlGenerator(kitchen_sink_path)
+    graphql = generator.serialize(classes=[input_class])
+
+    # check that the expected blocks are present
+    assert expected in graphql
```

Comment:
Just checking - is converting this to using the snapshot system TODO?

I think thats equivalent to these tests for string equality, neither of which are tests for correctness, but it lets the tests be updated as the generator changes and shows the whole schema.

Or is just the biolink model supposed to be snapshot?

Reply:
the version of biolink used for tests in this repo is frozen

Reply:
right right, i meant "should we replace these string comparison tests with a snapshot test or should we just use biolink as the snapshot test?" but i have said enough about this PR and withdraw my comment <3

Reply:
Snapshot testing added later on.